lib/screens/explore_screens/regions_screens/edit_region_event/edit_region_event_time_and_date.dart

```diff
@@ -0,0 +1,108 @@
+                child: Stack(
+                  children: [
+                    Column(
+                      children: [
+                        const SizedBox(height: 20),
+                        DateTimeRow(
+                          label: "Select Event Date",
+                          icon: const Icon(Icons.calendar_today_outlined),
+                          onWidgetTapped: () async {
+                            final DateTime? selected = await showDatePicker(
+                              context: context,
+                              initialDate: DateTime.now(),
+                              firstDate: DateTime.now(),
+                              lastDate: DateTime(2099),
+                            );
+                            // ignore: use_build_context_synchronously
+                            BlocProvider.of<EditRegionEventBloc>(context).add(OnSelectDateChanged(selected));
+                          },
+                          timeInfo: state.newEventDateAndTime != null
+                              ? DateFormat.yMMMMEEEEd().format(state.newEventDateAndTime!)
+                              : state.event.formattedCreatedTime,
+                        ),
+                        const SizedBox(height: 30),
+                        DateTimeRow(
+                          label: "Select Event Start Time",
+                          icon: const Icon(Icons.access_time),
+                          onWidgetTapped: () async {
+                            final TimeOfDay? picked = await showTimePicker(
+                              context: context,
+                              initialTime: const TimeOfDay(hour: 00, minute: 00),
+                            );
+                            // ignore: use_build_context_synchronously
+                            BlocProvider.of<EditRegionEventBloc>(context).add(OnStartTimeChanged(picked));
+                          },
+                          timeInfo: state.newEventStartTime != null
+                              ? "${state.newEventStartTime!.format(context)} - Starts"
+                              : state.event.formattedStartTime,
+                        ),
+                        const SizedBox(height: 30),
+                        DateTimeRow(
+                          label: "Select Event End Time",
+                          icon: const Icon(Icons.access_time),
+                          onWidgetTapped: () async {
+                            final TimeOfDay? picked = await showTimePicker(
+                              context: context,
+                              initialTime: const TimeOfDay(hour: 00, minute: 00),
+                            );
+                            // ignore: use_build_context_synchronously
+                            BlocProvider.of<EditRegionEventBloc>(context).add(OnEndTimeChanged(picked));
+                          },
+                          timeInfo: state.newEventEndTime != null
```

Comment:
move this logic out of the UI pls 

state.newEventEndTime != null
                              ? "${state.newEventEndTime!.format(context)} - Ends"
                              : state.event.formattedEndTime,
                              
                              
                              it could live in a `get` in the state

Reply:
same with other similar places

Reply:
fix